statsd/tests/test_statsd.py

```diff
@@ -0,0 +1,78 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import os
+import sys
+import time
+import pytest
+import requests
+import subprocess
+from datadog_checks.utils.common import get_docker_hostname
+from datadog_checks.statsd import StatsCheck
+
+CHECK_NAME = 'statsd'
+HOST = get_docker_hostname()
+PORT = 8126
+HERE = os.path.dirname(os.path.abspath(__file__))
+URL = "http://{}:{}".format(HOST, PORT)
+
+METRICS = [
+    'statsd.graphite.flush_length',
+    'statsd.messages.last_msg_seen',
+    'statsd.counters.count',
+    'statsd.graphite.last_exception',
+    'statsd.gauges.count',
+    'statsd.messages.bad_lines_seen',
+    'statsd.graphite.flush_time',
+    'statsd.graphite.last_flush',
+    'statsd.uptime',
+    'statsd.timers.count'
+]
+
+SERVICE_CHECKS = [
+    'statsd.is_up',
+    'statsd.can_connect'
+]
+
+
+@pytest.fixture
+def get_instance():
+    return {
+        'host': HOST,
+        'port': PORT,
+    }
+
+
+@pytest.fixture
+def aggregator():
+    from datadog_checks.stubs import aggregator
+    aggregator.reset()
+    return aggregator
+
+
+@pytest.fixture(scope="session")
+def spin_up_statsd():
+    env = os.environ
+    args = [
+        'docker-compose', '-f', os.path.join(HERE, 'compose', 'statsd.yaml')
+    ]
+    subprocess.check_call(args + ["up", "-d"], env=env)
+    sys.stdout.write("Waiting for Statsd instance to boot")
+    try:
+        res = requests.get(URL)
+        res.raise_for_status()
+    except Exception:
+        time.sleep(1)
+    yield
```

Comment:
I think what'll happen now is that even if the endpoint isn't up, the fixture will still `yield` to the tests without waiting and checking again if the endpoint is up.

You can do something like:
```
for _ in xrange(10):
    try:
        res = requests.get(URL)
        res.raise_for_status()
        break
    except Exception:
        time.sleep(1)
else:
    raise Exception("StatsD instance failed to boot")
```

But it's probably cleaner to create a `wait_for(...)` function.